clang/include/clang-c/CXString.h

```diff
@@ -46,6 +46,9 @@ typedef struct {
 
 /**
  * Retrieve the character data associated with the given string.
+ *
+ * The caller shouldn't free the returned string data, and the returned string
+ * data shouldn't be accessed after the \c CXString disposed.
```

Comment:
I think that this is confusing. We could be more explicit about releasing the resources though as you point out.

```suggestion
 * The returned data is a reference and now owned by the user. This data
 * is only valid while the `CXString` is valid. This function is similar
 * to `std::string::c_str()`.
```

Reply:
The returned data is owned by the user? I interpret it as the user takes the ownership. So there will be the user's responsibility to free the pointer. I don't think that's true.

Reply:
Whoops! That's my mistake - I meant to say `not`, not `now`. The data is the inner pointer like `c_str` is, and not meant to be modified by the user.

Reply:
Sounds great, but in your sentence, you may have to define what is "valid". (You said "while the `CXString` is valid.") 
@compnerd

Reply:
I'm not sure I have a better way to phrase it off hand. The validity here is referring to lifetimes in C/C++, which is normally described as "valid". See https://en.cppreference.com/w/cpp/string/basic_string/c_str as an example of the opposite (i.e. invalidation vs valid).

Reply:
> The validity here is referring to lifetimes in C/C++, which is normally described as \"valid\".

Great, but the string data is not valid during the whole lifetime of CXString. In opposite, it's only valid before we call the `clang_disposeString`.